ale/drivers/selene_drivers.py

```diff
@@ -520,33 +530,302 @@ def detector_start_sample(self):
+        be trusted.
+
+        Returns
+        -------
+        : float
+          Line exposure duration
+        """
+        # It's a list, but only sometimes.
+        # seems to depend on whether you are using the original zipped archives or
+        # if its downloaded from JAXA's image search:
+        # (https://darts.isas.jaxa.jp/planet/pdap/selene/product_search.html#)
+        try:
+            return self.label['IsisCube']['Instrument']['LineSamplingInterval'][0].value * 0.001 # Scale to seconds
+        except:
+            return self.label['IsisCube']['Instrument']['LineSamplingInterval'].value * 0.001  # Scale to seconds
+
+    @property
+    def detector_start_line(self):
+        return 1
+
+    @property
+    def detector_start_sample(self):
+        """
+        """
+        start_sample = 1
+        swath_mode = str(self.label["IsisCube"]["Instrument"]["SwathModeId"])
+        if (swath_mode == "FULL"):
+          start_sample = 1
+        elif (swath_mode == "NOMINAL"):
+          start_sample = 297
+        elif (swath_mode == "HALF"):
+          start_sample = 1172;
+        return start_sample - 0.5
+
+    @property
+    def focal2pixel_lines(self):
+        """
+        Calculated using 1/pixel pitch
+        Expects tc_id to be defined. This should be a string of the form
+        LISM_TC1 or LISM_TC2.
+
+        Returns
+        -------
+        : list
+          focal plane to detector lines
         """
+        pixel_size = spice.gdpool('INS{}_PIXEL_SIZE'.format(self.ikid), 0, 1)[0]
+        return [0, -1/pixel_size, 0]
+
+    @property
+    def focal2pixel_samples(self):
+        """
+        Calculated using 1/pixel pitch
+        Expects tc_id to be defined. This should be a string of the form
+        LISM_TC1 or LISM_TC2.
+
+        Returns
+        -------
+        : list
+          focal plane to detector samples
+        """
+        pixel_size = spice.gdpool('INS{}_PIXEL_SIZE'.format(self.ikid), 0, 1)[0]
+        return [0, 0, -1/pixel_size]
+
+    @property
+    def detector_center_line(self):
+        """
+        Returns the center detector line of the detector. Expects ikid to be
+        defined. This should be the NAIF integer ID code for the sensor.
+
+        We subtract 0.5 from the center line because as per the IK:
+        Center of the first pixel is defined as "1.0".
+
         Returns
         -------
         : int
-          ISIS sensor model version
+          The detector line of the principle point
+        """
+        return spice.gdpool('INS{}_CENTER'.format(self.ikid), 0, 2)[1] - 0.5
+
+    @property
+    def detector_center_sample(self):
+        """
+        Returns the center detector sample of the detector. Expects ikid to be
+        defined. This should be the NAIF integer ID code for the sensor.
+
+        We subtract 0.5 from the center sample because as per the IK:
+        Center of the first pixel is defined as "1.0".
```

Comment:
and ISIS needs to recenter to 0.5, correct?

Reply:
Yes, and knoten handles the conversion if are comparing between ISIS and USGSCSM